sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -597,6 +619,44 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
     """
   }
 
+  private[this] def codegenWriteArrayElemCode(et: DataType, ctx: CodegenContext): String = {
```

Comment:
It returns a function to write the array elements, maybe a better name is: `writeArrayToStringBuilderFunc`

Reply:
oh wait, the returned function is only called once, I think we don't need to make it a function, but just return the code, e.g.
```
def writeArrayToStringBuilder(ctx: CodegenContext, et: DataType, arr: String, builder: String): String
```

Reply:
`elementToString` needs a function because it's called twice.

Reply:
ok, I'll update soon.